### ISSUE TITLE
Package Aliasing

### ISSUE THREAD

**Issue by ian-weir** [oven-sh/bun#904]
### What is the problem this feature will solve?

If multiple versions of a dependency are required for a project aliasing the different version is very helpful to development. Aliasing allows your require/import statements to use the aliased name as well so it's clear in the code which version is being used.

### What is the feature you are proposing to solve the problem?

This is how yarn has solved it: https://classic.yarnpkg.com/en/docs/cli/add#toc-yarn-add-alias

**Comment by plibither8** [comment]
Bumping this issue, package aliasing would be super helpful!

**Comment by Jarred-Sumner** [comment]
Fixed by @alexlamsl in #1837

**Comment by ian-weir** [comment]
@Jarred-Sumner I'm not strictly sure it is unless I just have the configuration incorrect.

For instance if I have this in my package.json

```
    "docxtemplater": "3.26.4",
    "docxtemplater-legacy": "docxtemplater@3.19.7"
```

I get output like this from running `bun install`

error: package "docxtemplater-legacy" not found registry.yarnpkg.com/docxtemplater-legacy 404
error: docxtemplater-legacy@docxtemplater@3.19.7 failed to resolve

edit: I see I can the formatting incorrect as bun is mirroring npm so the correct format would be:

`"docxtemplater-legacy": "npm:docxtemplater@3.19.7",`
